### ISSUE TITLE
Allow `CircBox` to contain arbitrary `UnitID`

### ISSUE THREAD

**Issue by cqc-alec** [CQCL/tket#1057]
Add an optional `Dict[UnitID, UnitID]` argument to `Circuit.add_circbox()` to specify the wiring. (By default wiring is done in lexicographic order.)

**Comment by trvto** [comment]
I think this requires more effort than just changing the `add_circbox()` method. Currently, a `CircBox` can only be created from a simple `Circuit`

**Comment by cqc-alec** [comment]
> I think this requires more effort than just changing the `add_circbox()` method. Currently, a `CircBox` can only be created from a simple `Circuit`

Yes indeed, it will require changes to the `CircBox` implementation in the C++.
